api/main.py

```diff
@@ -216,8 +216,14 @@ async def get_nodes(request: Request, kind: str = "node"):
                 detail=f"Invalid request parameters: {msg}"
         )
 
-    translated_params = model.translate_fields(query_params)
-    return await db.find_by_attributes(model, translated_params)
+    try:
+        translated_params = model.translate_fields(query_params)
+        return await db.find_by_attributes(model, translated_params)
```

Comment:
Shouldn't this line be outside of the try-catch block?  The `ValueError` exceptions appear to only be meant for catching invalid input in other endpoint handlers.

Reply:
Operator mapping in the `db` can result in `ValueError` if the operator is not found.
That's why I added it to try-catch block.

Reply:
Right, but it's happening after validation.  So the validation is getting mixed up with the implementation of the logic to handle the API query.

I think this ValueError raised by the database class is not a validation issue, it's just an exception that happens while executing the code.  We could add some actual operator validation instead.  I think it's not a step in the right direction to be bundling validation with logic here.  But also I don't think we should be blocking this PR based on that, the whole design of exceptions and error propagation should be reviewed now we have more code and new use-cases that didn't exist when the endpoint handlers were first written.